handlers/addautohandler.go

```diff
@@ -17,16 +23,61 @@ func AddAutoHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	defer database.Close()
+	if r.Method == "GET" {
+		log.Printf("D: HTTP GET on /add-auto")
+		htmlHeadPath := "/templates/head.gohtml"
+		navBarPath := "/templates/navbar.gohtml"
+		templatePath := "/templates/addauto.gohtml"
 
-	htmlHeadPath := "/templates/head.gohtml"
-	navBarPath := "/templates/navbar.gohtml"
-	templatePath := "/templates/addauto.gohtml"
+		t, err := template.ParseFiles(templatePath, htmlHeadPath, navBarPath)
+		if err != nil {
+			log.Printf("E: Error processing template")
+		}
+		t.Execute(w, nil)
+	}
+	if r.Method == "POST" {
+		log.Printf("D: HTTP POST on /add-customer")
+		r.ParseForm()
+		var err error
+		var DateOfSale time.Time
 
-	t, err := template.ParseFiles(templatePath, htmlHeadPath, navBarPath)
+		_, has_param := r.Form["date"]
+		if has_param {
+			const (
+				layoutISO = "2006-01-02"
+			)
+			date := r.FormValue("date")
+			DateOfSale, err = time.Parse(layoutISO, date)
 
-	if err != nil {
-		log.Printf("E: Error processing template")
-	}
+			if err != nil {
+				panic(err)
+			}
+
+		}
+		Year, err := strconv.ParseInt(r.FormValue("year"), 10, 32)
+		if err != nil {
+			panic(err)
+		}
+		PurchasePrice, err := strconv.ParseFloat(r.FormValue("price"), 32)
+		if err != nil {
+			panic(err)
+		}
 
-	t.Execute(w, nil)
+		fmt.Println(DateOfSale)
+		newVehicle := models.Vehicle{
+			VIN:           r.FormValue("vin"),
+			Year:          int(Year),
+			Make:          r.FormValue("make"),
+			Model:         r.FormValue("model"),
+			PurchasePrice: int(PurchasePrice),
+			DateOfSale:    DateOfSale.String(),
+		}
+		err = utils.InsterAuto(newVehicle, database)
```

Comment:
I see that this is correctly named in sqlitequeries.go - do you mean to call this "InsertAuto"? that would match the schema for existing InsertCustomer etc.